prefab/prefab-publishing/mylibrary/build.gradle

```diff
@@ -39,14 +37,26 @@ android {
         }
     }
 
+    // Enable prefab intergrated inside AGP 4.1+
     buildFeatures {
         prefabPublishing true
     }
 
+    // Configure prefab for module headers
     prefab {
         mylibrary {
             headers "src/main/cpp/include"
         }
     }
+
+    // Avoid packing the unnecessary libraries into final AAR, for details
+    // refer to https://issuetracker.google.com/issues/168777344#comment5
+    // Note that if your AAR's Java code uses any C/C++ code support in the same AAR,
+    //      the workaround could not be used.
```

Comment:
This isn't really accurate. If the Java code uses libotherlibrary.so it's fine to exclude libmylibrary.so (and, frankly, if they're using libc++_shared in their JNI code their AAR is probably unusable: https://android.googlesource.com/platform/ndk/+/master/docs/user/middleware_vendors.md#for-java-middleware-with-jni-libraries).

How about:

Note that if your AAR also contains Java/Kotlin APIs, you should not exclude libraries that are used by those APIs.

Reply:
Right now, the "exclude" is a global switch: once turned on, all *.so files will be excluded ( otherlibrary and mylibrary ), which is very dangerous. this switch is not prefab specific. but for pure native AAR purpose, there is no java code, could use this workaround.

Reply:
the "exclude" is global: it will exclude any *.so files packed into AAR's "jni" folder: it is kind of dangerous, would break the AARs if those AARs do have some C++ code supporting their java code ( assume they have some Java API exported too ).  But for pure native AARs ( without any java code ), we could use this workaround.

Reply:
Still needs to be fixed.

Reply:
> the "exclude" is global: it will exclude any *.so files packed into AAR's "jni" folder

Not with the wildcard you've given. It only excludes the libraries that match the wildcard.

Reply:
yeah, make sense now. fixed by removing the second statement.